docs/design/pattern_matching.md

```diff
@@ -318,10 +300,15 @@ scrutinee.
 
 -   _proper-pattern_ ::= `var` _proper-pattern_
 
-A `var` pattern matches when its nested pattern matches. The type of the storage
-is the resolved type of the nested _pattern_. Any binding patterns within the
-nested pattern are reference binding patterns, and their bindings refer to
-portions of the corresponding storage rather than to the scrutinee.
+The scrutinee is expected to have the same type as the resolved type of the
+nested _proper-pattern_, and it is expected to be a runtime-phase owning
+ephemeral reference expression. The scrutinee expression is converted as needed
+to satisfy those expectations, and the `var` pattern takes ownership of the
+referenced object, promotes it to an owning _durable_ reference expression, and
+matches the nested _proper-pattern_ with it.
+
+The lifetime of the allocated object extends to the end of scope of the `var`
+pattern (that is the scope that any bindings declared within it would have).
```

Comment:
Notably: this means that for an owning ephemeral reference expression, we don't only require that we know statically that the object is complete, but we also require that we know *which* object it is (and that there's only one, and that the `var` is reached whenever the object is created), because we're going to modify properties of the original object (by extending its lifetime) at the point where we perform the `var` binding.